Task_5/classes/TaskCollection.js

```diff
@@ -0,0 +1,353 @@
+      console.error(error);
+      return false;
+    }
+  }
+
+  add(name, description, assignee, status, priority, isPrivate) {
+    try {
+      if (!this._isValidateUser()) {
+        throw new Error(ERRORS.userValidation);
+      }
+      if (!validateObj.name(name)) {
+        throw new Error(ERRORS.nameNotValidate);
+      }
+      if (!validateObj.description(description)) {
+        throw new Error(ERRORS.descriptionNotValidate);
+      }
+      if (!assignee) {
+        throw new Error(ERRORS.assigneeEmpty);
+      }
+      if (!validateObj.status(status)) {
+        throw new Error(ERRORS.statusNotValidate);
+      }
+      if (!validateObj.priority(priority)) {
+        throw new Error(ERRORS.priorityNotValidate);
+      }
+      if (!validateObj.isPrivate(isPrivate)) {
+        throw new Error(ERRORS.isPrivateNotValidate);
+      }
+
+      const task = new Task(
+        Helper.generateId(this._myCollection),
+        name,
+        description,
+        new Date(),
+        assignee,
+        status,
+        priority,
+        isPrivate,
+        []
+      );
+
+      this._myCollection.push(task);
+
+      Helper.showMessages(INFO.addSuccess);
+      return true;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  }
+
+  edit(
+    id,
+    name = null,
+    description = null,
+    assignee = null,
+    status = null,
+    priority = null,
+    isPrivate = false
+  ) {
+    try {
+      if (!this._isValidateUser()) {
+        throw new Error(ERRORS.userValidation);
+      }
+      if (arguments.length <= 1) {
+        throw new Error(ERRORS.countAgrumentsNotValidate);
+      }
+
+      if (!Helper.isValidTypeId(id)) {
+        throw new Error(ERRORS.invalidValue);
+      }
+
+      if (!this._findTaskById(id)) {
+        throw new Error(ERRORS.taskNotFound);
+      }
+
+      if (!this._isValidUserInTask(id)) {
+        throw new Error(ERRORS.userValidation);
+      }
+
+      const task = this.get(id);
+
+      if (name) {
+        if (!validateObj.name(name)) {
+          throw new Error(ERRORS.nameNotValidate);
+        }
+        task.name = name;
+      }
+
+      if (description) {
+        if (!validateObj.description(description)) {
+          throw new Error(ERRORS.descriptionNotValidate);
+        } else {
+          task.description = description;
+        }
+      }
+
+      if (assignee) {
+        if (!validateObj.description(assignee)) {
+          throw new Error(ERRORS.assigneeEmpty);
+        } else {
+          task.assignee = assignee;
+        }
+      }
+
+      if (status) {
+        if (!validateObj.description(status)) {
+          throw new Error(ERRORS.statusNotValidate);
+        } else {
+          task.status = status;
+        }
+      }
+
+      if (priority) {
+        if (!validateObj.description(priority)) {
+          throw new Error(ERRORS.priorityNotValidate);
+        } else {
+          task.priority = priority;
+        }
+      }
+
+      if (isPrivate) {
```

Comment:
if you modify task with isPrivate = false you dont chancge this field. Modify check